### ISSUE TITLE
Reproduce on WLASL100 problem

### ISSUE THREAD

**Issue by lucasnhandang** [ZechengLi19/Uni-Sign#31]
Hi, I have downloaded the wlasl_rgb_pose_islr.pth (pretrained weights) file from Hugging Face and was able to load the data successfully. I have also verified that all data paths are correctly configured.

**When evaluating on WLASL2000** using the same configuration as described in paper, I obtained comparable results (with top-1 accuracy even slightly higher):

<img width="1576" height="479" alt="Image" src="https://github.com/user-attachments/assets/1dcebf2c-ecf4-479c-b819-dacb6eb6b425" />

However, **when evaluating on WLASL100**, the performance is noticeably lower than expected:

<img width="1575" height="456" alt="Image" src="https://github.com/user-attachments/assets/2f3d72c3-4875-4bee-bec3-d93cd5b68a3a" />

I am currently unsure what might be causing this discrepancy. I would greatly appreciate any insights or suggestions regarding potential reasons or things I might have overlooked.

Thank you very much for your time and support!

**Comment by ZechengLi19** [comment]
@lucasnhandang Hi.

It’s great to hear that you were able to reproduce the WLASL2000 results successfully.

Regarding the lower performance on WLASL100, the key point is that the pretrained weights you are using are trained for the WLASL2000 setting. Therefore, if you would like to properly evaluate performance on WLASL100, you will need to fine-tune the model on the WLASL100 training set. In practice, this means initializing from the ASL Stage1 or Stage2 pretrained weights and then performing fine-tuning specifically on WLASL100 before evaluation. Directly applying WLASL2000-trained weights to WLASL100 without adaptation will typically lead to suboptimal performance due to differences in label space and data distribution.

That said, if you would still like to evaluate using the WLASL2000 weights directly, one possible workaround is to modify the decoding process. For example, you could allow the model to generate multiple candidate words and then select the highest-scoring prediction among those that belong to the WLASL100 vocabulary. This might help improve performance to some extent, although I have not personally verified this approach.

**Comment by lucasnhandang** [comment]
Thank you very much for your detailed explanation, I really appreciate it!
